### ISSUE TITLE
Improve how component value clamping is represented in the "latest at" dataframe

### ISSUE THREAD

**Issue by abey79** [rerun-io/rerun#6910]
After #6894, components which have a lower-than-max instance count have their last value "duplicated" with a dimmer style, to represent the clamplng:

<img width="794" alt="image" src="https://github.com/user-attachments/assets/db8b7471-9bee-4c67-b6a7-3f649a5ba087">

It is not entirely clear that this is a good design:
- For starters, we're displaying data where there is none, which may contrary to the basic idea of the data table.
- The actual clamping behaviour is very much visualiser dependant. It's often done against some primary/required component (e.g. `Positions3D`), and not "whichever component happens to have the most instances". In some case (e.g. mesh), clamping makes no sense whatsoever. There exists other cases with different behaviour. So any clamping behaviour shown in the dataframe view has high chances of not being actually representative of what happens elsewhere.

For example, see here `Position3D` being clamped against `TriangleIndices`, which makes no sense:

<img width="1423" alt="image" src="https://github.com/user-attachments/assets/4ba42306-029c-4a14-95b3-a4b41b7d9159">

#### Possible avenues

- Have some archetype-level metadata to specify the actual clamping behaviour, and the possibility to somehow choose an archetype "point of view" for the dataframe (which, as data filtering UI goes, would be pretty interesting beyond the mere question of clamping display).
- Have some heuristics to, e.g., only clamp component with length = 1 (aka "splat), as opposed to clamping component with different, >1 lengths as for meshes.

#### Design

Design proposal by @gavrelina to improve how clamped value are represented:

https://github.com/user-attachments/assets/4b63aaaa-bd99-4de0-84c7-86344e68b223

**Comment by abey79** [comment]
- Superseded by  #7066
